src/renderer_mtl.mm

```diff
@@ -3510,20 +3545,37 @@ void writeString(bx::WriterI* _writer, const char* _str)
 	{
 		if (NULL == m_drawableTexture)
 		{
+#if BX_PLATFORM_VISIONOS
+			m_frame = cp_layer_renderer_query_next_frame(m_layerRenderer);
+			if (m_frame) {
+				m_drawable =  cp_frame_query_drawable(m_frame);
+			}
+#else // BX_PLATFORM_VISIONOS
 			m_drawable = m_metalLayer.nextDrawable;
+#endif // BX_PLATFORM_VISIONOS
 			if (m_drawable != NULL)
 			{
+#if BX_PLATFORM_VISIONOS
+				m_drawableTexture = cp_drawable_get_color_texture(m_drawable, 0);
+#else // BX_PLATFORM_VISIONOS
 				m_drawableTexture = m_drawable.texture;
-				retain(m_drawableTexture);
 				retain(m_drawable); // keep alive to be useable at 'flip'
+#endif // BX_PLATFORM_VISIONOS
+				retain(m_drawableTexture);
 			}
 			else
 			{
 				TextureDescriptor desc = s_renderMtl->m_textureDescriptor;
 				desc.textureType = MTLTextureType2D;
+#if !BX_PLATFORM_VISIONOS
 				desc.pixelFormat = m_metalLayer.pixelFormat;
 				desc.width  = m_metalLayer.drawableSize.width;
 				desc.height = m_metalLayer.drawableSize.height;
+#else // BX_PLATFORM_VISIONOS
+				desc.pixelFormat = MTLPixelFormatBGRA8Unorm_sRGB;
+				desc.width  = 2732;
+				desc.height = 2048;
+#endif // BX_PLATFORM_VISIONOS
```

Comment:
Is this really impossible to obtain via `drawableSize`?!

Reply:
Yeah, it's not possible, unfortunately, you can only obtain the size from the drawable which in this case is null. I also noticed that this code is called only when we de-initialize bgfx and removing it didn't change anything, so it's not there anymore.